cli/info.rs

```diff
@@ -1,404 +1,232 @@
+  pub files: ModuleInfoMap,
+  #[serde(skip_serializing)]
+  pub info: ModuleInfo,
+  pub local: PathBuf,
+  pub map: Option<PathBuf>,
+  #[serde(serialize_with = "serialize_specifier")]
+  pub module: ModuleSpecifier,
+  pub total_size: usize,
 }
 
-impl std::fmt::Display for ModuleDepInfo {
-  fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-    f.write_fmt(format_args!("{} {}\n", colors::bold("local:"), self.local))?;
-    f.write_fmt(format_args!(
-      "{} {}\n",
-      colors::bold("type:"),
-      self.file_type
-    ))?;
+impl fmt::Display for ModuleGraphInfo {
+  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+    writeln!(
+      f,
+      "{} {}",
+      colors::bold("local:"),
+      self.local.to_string_lossy()
+    )?;
+    writeln!(f, "{} {}", colors::bold("type:"), self.file_type)?;
     if let Some(ref compiled) = self.compiled {
-      f.write_fmt(format_args!(
-        "{} {}\n",
+      writeln!(
+        f,
+        "{} {}",
         colors::bold("compiled:"),
-        compiled
-      ))?;
+        compiled.to_string_lossy()
+      )?;
     }
     if let Some(ref map) = self.map {
-      f.write_fmt(format_args!("{} {}\n", colors::bold("map:"), map))?;
+      writeln!(f, "{} {}", colors::bold("map:"), map.to_string_lossy())?;
     }
-
-    f.write_fmt(format_args!(
-      "{} {} unique {}\n",
+    writeln!(
+      f,
+      "{} {} unique {}",
       colors::bold("deps:"),
       self.dep_count,
       colors::gray(&format!(
         "(total {})",
-        human_size(self.deps.total_size.unwrap_or(0) as f64),
+        human_size(self.info.total_size.unwrap_or(0) as f64)
       ))
-    ))?;
-    f.write_fmt(format_args!(
-      "{} {}\n",
-      self.deps.name,
-      colors::gray(&format!("({})", human_size(self.deps.size as f64)))
-    ))?;
-
-    for (idx, dep) in self.deps.deps.iter().enumerate() {
-      print_file_dep_info(&dep, "", idx == self.deps.deps.len() - 1, f)?;
+    )?;
+    writeln!(f)?;
+    writeln!(
+      f,
+      "{} {}",
+      self.info.name,
+      colors::gray(&format!("({})", human_size(self.info.size as f64)))
+    )?;
+
+    let dep_count = self.info.deps.len();
+    for (idx, dep) in self.info.deps.iter().enumerate() {
+      dep.write_info(f, &"", idx == dep_count - 1)?;
     }
 
     Ok(())
   }
 }
 
-/// A dependency tree of the basic module information.
-///
-/// Constructed from a `ModuleGraph` and `ModuleSpecifier` that
-/// acts as the root of the tree.
-#[derive(Serialize)]
+/// Represents a unique dependency within the graph of the the dependencies for
+/// a given module.
+#[derive(Debug, Serialize, Eq, PartialEq)]
 #[serde(rename_all = "camelCase")]
-struct FileInfoDepTree {
-  name: String,
-  size: usize,
-  total_size: Option<usize>,
-  deps: Vec<FileInfoDepTree>,
+pub struct ModuleInfo {
+  pub deps: Vec<ModuleInfo>,
+  #[serde(serialize_with = "serialize_specifier")]
+  pub name: ModuleSpecifier,
+  pub size: usize,
+  pub total_size: Option<usize>,
```

Comment:
I'd try to avoid superfluous renames/reordering of structs where possible to keep the diff small. I can't tell if this is related to the module graph or is just a better name/ordering.

Reply:
Again, it seems totally non-sensical not to clean up stuff... otherwise it never gets cleaned up.